lib/usd/translators/shading/usdBlinnReader.cpp

```diff
@@ -0,0 +1,134 @@
+#include <mayaUsd/fileio/translators/translatorUtil.h>
+#include <mayaUsd/fileio/utils/readUtil.h>
+#include <mayaUsd/utils/util.h>
+
+#include <pxr/base/tf/diagnostic.h>
+#include <pxr/base/tf/staticTokens.h>
+#include <pxr/base/tf/token.h>
+#include <pxr/base/vt/value.h>
+#include <pxr/pxr.h>
+#include <pxr/usd/sdf/path.h>
+#include <pxr/usd/sdf/types.h>
+#include <pxr/usd/sdf/valueTypeName.h>
+#include <pxr/usd/usdShade/input.h>
+#include <pxr/usd/usdShade/output.h>
+#include <pxr/usd/usdShade/shader.h>
+#include <pxr/usd/usdShade/tokens.h>
+
+#include <maya/MFnBlinnShader.h>
+#include <maya/MFnDependencyNode.h>
+#include <maya/MObject.h>
+#include <maya/MPlug.h>
+#include <maya/MStatus.h>
+
+#include <basePxrUsdPreviewSurface/usdPreviewSurface.h>
+
+PXR_NAMESPACE_OPEN_SCOPE
+
+/// Shader writer for importing UsdPreviewSurface to Maya's lambert material nodes
+class PxrUsdTranslators_BlinnReader : public PxrUsdTranslators_LambertReader {
+    typedef PxrUsdTranslators_LambertReader _BaseClass;
+
+public:
+    PxrUsdTranslators_BlinnReader(const UsdMayaPrimReaderArgs&);
+
+    static ContextSupport CanImport(const UsdMayaJobImportArgs& importArgs);
+
+    /// Get the name of the Maya shading attribute that corresponds to the
+    /// USD attribute named \p usdAttrName.
+    virtual TfToken GetMayaNameForUsdAttrName(const TfToken& usdAttrName) const;
+
+protected:
+    /// What is the Maya node type name we want to convert to:
+    const TfToken& _GetMayaNodeTypeName() const override;
+
+    void _OnReadAttribute(const TfToken& mayaAttrName, MFnDependencyNode& shaderFn) const override;
+};
+
+PXRUSDMAYA_REGISTER_SHADER_READER(UsdPreviewSurface, PxrUsdTranslators_BlinnReader);
+
+// clang-format off
+TF_DEFINE_PRIVATE_TOKENS(
+    _tokens,
+
+    // Maya material nodes attribute names
+    (eccentricity)
+    (specularColor)
+);
+// clang-format on
+
+PxrUsdTranslators_BlinnReader::PxrUsdTranslators_BlinnReader(const UsdMayaPrimReaderArgs& readArgs)
+    : PxrUsdTranslators_LambertReader(readArgs)
+{
+}
+
+/* static */
+UsdMayaShaderReader::ContextSupport
+PxrUsdTranslators_BlinnReader::CanImport(const UsdMayaJobImportArgs& importArgs)
+{
+    return importArgs.shadingConversion == UsdMayaShadingConversionTokens->blinn
+        ? ContextSupport::Supported
+        : ContextSupport::Unsupported;
+}
+
+const TfToken& PxrUsdTranslators_BlinnReader::_GetMayaNodeTypeName() const
+{
+    return UsdMayaShadingConversionTokens->blinn;
+}
+
+void PxrUsdTranslators_BlinnReader::_OnReadAttribute(
+    const TfToken&     mayaAttrName,
+    MFnDependencyNode& shaderFn) const
+{
+    if (mayaAttrName == _tokens->specularColor) {
+        MFnBlinnShader blinnFn;
+        blinnFn.setObject(shaderFn.object());
+        MColor color(blinnFn.specularColor());
+        float  scale(blinnFn.specularRollOff());
+        color /= scale;
+        blinnFn.setSpecularColor(color);
+        blinnFn.setSpecularRollOff(1.0f);
```

Comment:
On export we scale the specular color by the rollOff. So on import we must set the rollOff to 1.

Reply:
I mentioned this in `usdMaterialReader.cpp` where `_OnReadAttribute()` gets called, but this code actually looks kind of odd until you realize that this is a "prep" step **before** we apply the values from USD, since we wouldn't have read a "rolloff" attribute, for example.